test/extended/operators/daemon_set.go

```diff
@@ -0,0 +1,73 @@
+	"k8s.io/apimachinery/pkg/labels"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[sig-arch] Managed cluster", func() {
+	oc := exutil.NewCLIWithoutNamespace("operator-daemonsets")
+
+	// Daemonsets shipped with the platform must be able to upgrade without disruption to workloads.
+	// Daemonsets that are in the data path must gracefully shutdown and redirect workload traffic, or
+	// mitigate outage by holding requests briefly (very briefly!). Daemonsets that are part of control
+	// plane (anything that changes state) must already be able to be upgraded without disruption by
+	// having any components that call them retry when unavailable up to a specific duration. Therefore
+	// all workloads must individually be graceful, and so a core daemonset can upgrade up to 10% of
+	// pods at a time. In the future, we may allow this percentage to be tuned via a global config, and
+	// this test would enforce that.
+	//
+	// Currently only applies to daemonsets that don't explicitly target the control plane.
+	g.It("should only include cluster daemonsets that have maxUnavailable update of 10 percent", func() {
+		// iterate over the references to find valid images
+		daemonSets, err := oc.KubeFramework().ClientSet.AppsV1().DaemonSets("").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			e2e.Failf("unable to list daemonsets: %v", err)
+		}
+
+		// daemonsets that have an explicit exception, every entry here must have a bug associated
+		// e.g. "openshift-sdn/foo": "link to bug"
+		skipDaemonSets := map[string]string{
+			"openshift-multus/multus":                            "https://bugzilla.redhat.com/show_bug.cgi?id=1933159",
+			"openshift-sdn/sdn":                                  "https://bugzilla.redhat.com/show_bug.cgi?id=1933173",
+			"openshift-sdn/ovs":                                  "https://bugzilla.redhat.com/show_bug.cgi?id=1933174",
+			"openshift-cluster-csi-drivers":                      "https://bugzilla.redhat.com/show_bug.cgi?id=1933184",
+			"openshift-image-registry/node-ca":                   "https://bugzilla.redhat.com/show_bug.cgi?id=1933180",
+			"openshift-network-diagnostics/network-check-target": "https://bugzilla.redhat.com/show_bug.cgi?id=1933179",
+		}
+
+		var invalidDaemonSets []string
+		for _, ds := range daemonSets.Items {
+			if !strings.HasPrefix(ds.Namespace, "openshift-") {
+				continue
+			}
+			if ds.Spec.Selector != nil {
+				selector := labels.SelectorFromSet(ds.Spec.Template.Spec.NodeSelector)
+				if !selector.Empty() && selector.Matches(labels.Set(map[string]string{"node-role.kubernetes.io/master": ""})) {
+					continue
+				}
+			}
+			if bug, ok := skipDaemonSets[fmt.Sprintf("%s/%s", ds.Namespace, ds.Name)]; ok {
+				e2e.Logf("Skipping daemonset %s/%s from consideration: %s", ds.Namespace, ds.Name, bug)
+				continue
+			}
+			switch {
+			case ds.Spec.UpdateStrategy.RollingUpdate == nil,
+				ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable == nil:
```

Comment:
None of these today:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-gcp-4.8/1364411741282242560/artifacts/e2e-gcp/daemonsets.json | jq -r '.items[] | .metadata.namespace + " " + .metadata.name + " " + (.spec.template.spec.nodeSelector | tostring) + " " + (.spec.updateStrategy | tostring)' | grep -v RollingUpdate
...no hits...
```

Should we make them illegal unless folks talk to you about softening the CI guard?

Reply:
oh, good point.  thanks, i screwed that up in my head

"daemonsets must have a rolling update 10% strategy"

Reply:
This thread can be marked resolved, now that this is an error case.